src/main/java/net/sacredlabyrinth/phaed/simpleclans/managers/SettingsManager.java

```diff
@@ -115,20 +128,63 @@ private void warnAboutPluginDependencies() {
+        if (discordSrv == null && is(DISCORDCHAT_ENABLE)) {
             plugin.getLogger().warning("DiscordChat can't be initialized, please, install DiscordSRV.");
         }
     }
 
     public Locale getLanguage() {
-        String language = getString(LANGUAGE);
-        String[] split = language.split("_");
+        return LOCALE_CACHE.computeIfAbsent(getString(LANGUAGE), k -> {
+            String lang = getString(LANGUAGE);
 
-        if (split.length == 2) {
-            return new Locale(split[0], split[1]);
-        }
+            if (lang == null || lang.isEmpty()) {
+                lang = LANGUAGE.defaultValue.toString();
+            }
+            String[] langParts = lang.split("_");
+
+            if (langParts.length != 2) {
+                plugin.getLogger().warning(String.format("Invalid language: %s", lang));
+                plugin.getLogger().warning(String.format("Using default language: %s", LANGUAGE.defaultValue));
+                lang = LANGUAGE.defaultValue.toString();
+                langParts = lang.split("_");
+            }
+
+            return new Locale(langParts[0], langParts[1]);
+        });
+    }
+
+    public DecimalFormat getDecimalFormat() {
+        return FORMAT_CACHE.computeIfAbsent(getString(ECONOMY_DECIMAL_FORMAT_PATTERN), k -> {
+            DecimalFormatSymbols symbols = new DecimalFormatSymbols(getLanguage());
+
+            try {
+                return new DecimalFormat(getString(ECONOMY_DECIMAL_FORMAT_PATTERN), symbols);
+            } catch (IllegalArgumentException e) {
+                plugin.getLogger().warning(String.format("Invalid decimal-format-pattern: %s",
+                        getString(ECONOMY_DECIMAL_FORMAT_PATTERN)));
+                plugin.getLogger().warning(String.format("Using default decimal format pattern: %s",
+                        ECONOMY_DECIMAL_FORMAT_PATTERN.defaultValue));
+
+                return new DecimalFormat(ECONOMY_DECIMAL_FORMAT_PATTERN.defaultValue.toString(), symbols);
+            }
+        });
+    }
 
-        return new Locale(language);
+    public DateTimeFormatter getDateTimeFormatter() {
+        return DATE_CACHE.computeIfAbsent(getString(DATE_PATTERN), k -> {
+            try {
+                return DateTimeFormatter.ofPattern(getString(DATE_PATTERN))
+                        .withLocale(getLanguage())
+                        .withZone(ZoneId.systemDefault());
+            } catch (IllegalArgumentException e) {
+                plugin.getLogger().warning(String.format("Invalid date-time-format-pattern: %s", getString(DATE_PATTERN)));
+                plugin.getLogger().warning(String.format("Using default date pattern: %s", DATE_PATTERN.defaultValue));
+
+                return DateTimeFormatter.ofPattern(DATE_PATTERN.defaultValue.toString())
+                        .withLocale(getLanguage())
+                        .withZone(ZoneId.systemDefault());
+            }
+        });
```

Comment:
```java
    private <T> T cacheField(HashMap<String, T> hashMap, ConfigField field, Function<String, T> mapFunction) {
        return hashMap.computeIfAbsent(field.name(), mapFunction);
    }
```

Then:
```java
public DateTimeFormatter getDateTimeFormatter() {
  return cacheField(DATE_CACHE, DATE_PATTERN, ...);
}
```

What do you think?

Reply:
I think, this method can be used in SettingsManager#getString() to cache String values (and other types) as well.
But this isn't the purpose of this PR.

Reply:
`HashMap<String, T> hashMap` could be replaced with `HashMap<ConfigField, T> hashMap`. 🤔